src/main/java/io/phasetwo/keycloak/magic/auth/util/AuthenticatorSharedUtils.java

```diff
@@ -0,0 +1,21 @@
+package io.phasetwo.keycloak.magic.auth.util;
+
+import org.keycloak.authentication.AuthenticationFlowContext;
+import org.keycloak.models.AuthenticatorConfigModel;
+
+import java.util.Map;
+
+public final class AuthenticatorSharedUtils {
+    public static boolean is(AuthenticationFlowContext context, String propName, boolean defaultValue) {
+        AuthenticatorConfigModel authenticatorConfig = context.getAuthenticatorConfig();
+        if (authenticatorConfig == null) return defaultValue;
+
+        Map<String, String> config = authenticatorConfig.getConfig();
+        if (config == null) return defaultValue;
+
+        String v = config.get(propName);
+        if (v == null || "".equals(v)) return defaultValue;
```

Comment:
Prefer the guava method https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/base/Strings.html#isNullOrEmpty(java.lang.String)

Reply:
Did not know them, very nice. Changed to it.